dbms/src/Functions/randomASCII.cpp

```diff
@@ -0,0 +1,134 @@
+#include <Functions/IFunctionImpl.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <Columns/ColumnString.h>
+#include <Columns/ColumnVector.h>
+#include <Columns/ColumnConst.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <IO/WriteBufferFromVector.h>
+#include <IO/WriteHelpers.h>
+#include <Common/formatReadable.h>
+#include <Common/typeid_cast.h>
+#include <type_traits>
+
+#include <random>
+#include <iostream>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_COLUMN;
+}
+
+class FunctionRandomASCII : public IFunction
+{
+public:
+    static constexpr auto name = "randomASCII";
+    static FunctionPtr create(const Context &) { return std::make_shared<FunctionRandomASCII>(); }
+
+    String getName() const override
+    {
+        return name;
+    }
+
+    size_t getNumberOfArguments() const override { return 1; }
```

Comment:
We can also support additional "disambiguation" arguments, similar to `rand` function.